src/app/chat/chat-form/chat-form.component.spec.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  beforeEachProviders,
+  inject,
+  it
+} from "@angular/core/testing";
+import { Control } from "@angular/common";
+
+import { ChatFormComponent } from "./chat-form.component";
+
+// [TODO]: Move to action-service/action-service.mock.ts when ActionService mirated to Angular 2
+import { provide, Provider } from "@angular/core";
+class MockActionService {
+  constructor() { }
+  public writeChatMessage(text: string): void { }
+  public getProvider(): Provider {
+    return provide("ActionService", {useValue: this});
+  }
+}
+// end TODO
+
+describe("ChatForm", () => {
+  let mockActionService: MockActionService;
+
+  beforeEachProviders(() => {
+    mockActionService = new MockActionService();
+    return [
+      ChatFormComponent,
+      mockActionService.getProvider()
+    ];
+  });
+
+  it("should have chatForm defined", inject([ ChatFormComponent ], (chatForm) => {
+    expect(chatForm.chatForm).toBeDefined();
+  }));
+
+  it("should have control text defined", inject([ ChatFormComponent ], (chatForm) => {
+    let field: string = "text";
+
+    expect(chatForm.chatForm.controls[field]).toBeDefined();
+  }));
+
+  it("should have control text defined as null", inject([ ChatFormComponent ], (chatForm) => {
+    expect(chatForm.chatForm.value.text).toBe(null);
+  }));
+
+  it("should reset text value on submit", inject([ ChatFormComponent ], (chatForm) => {
+    let field: string = "text";
+
+    (<Control>chatForm.chatForm.controls[field]).updateValue("manual input");
+    chatForm.submit();
+
+    expect(chatForm.chatForm.value.text).toBe(null);
+  }));
+
+  it("should call ActionService.writeChatMessage on submit", inject([ ChatFormComponent ], (chatForm) => {
+    let field: string = "text";
+    spyOn(mockActionService, "writeChatMessage");
+
+    (<Control>chatForm.chatForm.controls[field]).updateValue("manual input");
+    chatForm.submit();
+
+    expect(mockActionService.writeChatMessage).toHaveBeenCalledWith("manual input");
+  }));
+
+});
```

Comment:
For this whole file I would like to suggest this alternative
```typescript
import {
  beforeEach, 
  beforeEachProviders,
  describe,
  expect,
  inject,
  it
} from "@angular/core/testing";

import { Control } from "@angular/common";
    
declare const jasmine;
declare const spyOn;
  
import { ChatFormComponent } from "./chat-form.component";

class MockActionService {
  public writeChatMessage(text: string): void { }
}

describe("ChatForm", () => {
  beforeEachProviders(() => {
    this.actionService = new MockActionService();

    return [
      {provide: ChatFormComponent, useClass: ChatFormComponent},
      {provide: "ActionService", useValue: this.actionService}
    ];
  });

  beforeEach(inject([ ChatFormComponent ], (chatForm)  => {
    this.chatForm = chatForm;
    this.ctrlGroup = chatForm.chatForm.value;
    this.textField = <Control>this.chatForm.chatForm.controls["text"]
  }));

  it("should start with an empty text", () => {
    expect(this.ctrlGroup.text).toBe(null);
    expect(this.textField.value).toBe(null);
  });

  it("should reset text value on submit", () => {
    this.textField.updateValue("manual input");
    this.chatForm.submit();

    expect(this.ctrlGroup.text).toBe(null);
    expect(this.textField.value).toBe(null);
  });

  it("should use ActionService.writeChatMessage to send the message", () => {
    spyOn(this.actionService, "writeChatMessage");

    this.textField.updateValue("manual input");
    this.chatForm.submit();

    expect(this.actionService.writeChatMessage).toHaveBeenCalledWith("manual input");
  });
});
```

Main changes:

 * Declare the injection only once, using `beforeEach`
 * Using the `this` object provided by Jasmine (a.k.a. `userContext`) not only to DRY the injection but also to have reusable meaningful names.
 * Focus in the behavior, not in the implementation. That means fewer tests with high-level descriptive (hopefully) texts.

In general, I find my version to be more readable. @magarcia, @imobachgs What do you guys think?